src/gui/search/searchtab.h

```diff
@@ -32,50 +32,86 @@
 #define SEARCHTAB_H
 
 #include <QWidget>
+#include "searchsortmodel.h"
+
+#define ENGINE_URL_COLUMN 4
+#define URL_COLUMN 5
 
 class QLabel;
 class QTreeView;
 class QHeaderView;
 class QStandardItemModel;
-class QSortFilterProxyModel;
-class QModelIndex;
 class QVBoxLayout;
 
 class SearchSortModel;
 class SearchListDelegate;
 class SearchWidget;
 
+namespace Ui
+{
+    class SearchTab;
+}
+
 class SearchTab: public QWidget
 {
     Q_OBJECT
 
 public:
+
+    enum NameFilteringMode
+    {
+        Everywhere,
+        OnlyNames
+    };
+
+    Q_ENUMS(NameFilteringMode)
+
     explicit SearchTab(SearchWidget *m_parent);
+    ~SearchTab();
 
-    QLabel* getCurrentLabel() const;
     QStandardItemModel* getCurrentSearchListModel() const;
-    QSortFilterProxyModel* getCurrentSearchListProxy() const;
+    SearchSortModel* getCurrentSearchListProxy() const;
     QTreeView* getCurrentTreeView() const;
     QHeaderView* header() const;
-    QString status() const;
 
     bool loadColWidthResultsList();
     void setRowColor(int row, QString color);
-    void setStatus(const QString &value);
+
+    enum class Status
+    {
+        Ongoing,
+        Finished,
+        Error,
+        Aborted,
+        NoResults
+    };
+
+    void setStatus(Status value);
+    Status status() const;
+
+public slots:
+    void updateResultsCount();
```

Comment:
The 'slot` is leftover. It used to be a receiver of the model signal. I'll remove that. But it has to be public because `SearchWidget` calls it too.